src/lib.rs

```diff
@@ -200,6 +201,208 @@ impl ReturnMode {
     }
 }
 
+#[derive(Debug, PartialEq)]
+pub enum SegmentationUpidType {
+    NotUsed,
+    UserDefinedDeprecated,
+    ISCIDeprecated,
+    AdID,
+    UMID,
+    ISANDeprecated,
+    ISAN,
+    TID,
+    TI,
+    ADI,
+    EIDR,
+    ATSC,
+    MPU,
+    MID,
+    ADS,
+    URI,
+    Reserved
```

Comment:
Including the specific id value within this variant will provide the best information to calling code

```suggestion
    Reserved(u8)
```

Reply:
Yes, this is far more sensible. Updated.